src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -163,6 +187,30 @@ async def infer_python_dependencies_via_imports(
         Get(PythonModuleOwners, PythonModule(imported_module))
         for imported_module in detected_imports
     )
+
+    unowned_dependency_behavior = python_infer_subsystem.options.unowned_dependency_behavior
+    if unowned_dependency_behavior is not UnownedDependencyUsage.DoNothing:
+        unowned_imports = sorted(
+            imp
+            for owners, imp in zip(owners_per_import, detected_imports)
+            if not owners.unambiguous and imp.split(".")[0] not in DEFAULT_UNOWNED_DEPENDENCIES
+        )
+        if unowned_imports:
+            messaging = (
+                # @TODO: I don't like the manual tabbing here.
```

Comment:
Opening comment for TODO.

I want a nice UI, any ideas?

Reply:
The ambiguity warnings below are rendered as complete log entries per warning... which is fine for warnings, and parses a little bit more easily. But I suppose that for the error case you would want a compound error so that you see all (or at least more) of them?

It's maybe slightly more awkward, but another option would be to log the errors at `logger.error` with line-per-error, and wait to raise the exception until after the loop, or even until the end of the function?

Reply:
I love this idea! :raised_hands:

Reply:
On the phone, so have a hard time making out what's what.. but there's the `bullet_list()` too, in utils (or some such, forgot the exact naming and place)

Reply:
Hopefully it won't come to that  and the logger-idea works out well. :pray:

Reply:
Logger looks good!

```
16:27:04.97 [ERROR] The following imports have no unambiguous owner:

  * humanize

If you are using [python-setup].requirement_constraints, consider adding the relevant package.
Otherwise consider specifying a python_requirement target as a dependency.
See https://www.pantsbuild.org/v2.8/docs/python-third-party-dependencies
16:27:04.97 [ERROR] Exception caught: (pants.engine.internals.scheduler.ExecutionError)
  File "/home/joshuacannon/work/pants/src/python/pants/bin/local_pants_runner.py", line 235, in _run_inner
    return self._perform_run(goals)
  File "/home/joshuacannon/work/pants/src/python/pants/bin/local_pants_runner.py", line 174, in _perform_run
    return self._perform_run_body(goals, poll=False)
  File "/home/joshuacannon/work/pants/src/python/pants/bin/local_pants_runner.py", line 191, in _perform_run_body
    return self.graph_session.run_goal_rules(
  File "/home/joshuacannon/work/pants/src/python/pants/init/engine_initializer.py", line 134, in run_goal_rules
    exit_code = self.scheduler_session.run_goal_rule(
  File "/home/joshuacannon/work/pants/src/python/pants/engine/internals/scheduler.py", line 548, in run_goal_rule
    self._raise_on_error([t for _, t in throws])
  File "/home/joshuacannon/work/pants/src/python/pants/engine/internals/scheduler.py", line 516, in _raise_on_error
    raise ExecutionError(

Exception message: 1 Exception encountered:

  UnownedDependencyError: One or more unowned dependencies detected. Check logs for more details.

```

Reply:
Yea, looks good! Sorry for the extra noise in there... that's not supposed to be rendering by default. Fixed in #13539.